app/models/event.rb

```diff
@@ -0,0 +1,16 @@
+# Event Model
+class Event < ApplicationRecord
+  include SessionHelper
+
+  belongs_to :user
+
+  scope :upcoming, -> { where('date > ?', DateTime.now.to_date) }
+  scope :past, -> { where('date < ?', DateTime.now.to_date) }
+
+  has_many :EventAttendance
+  has_many :attended_users, through: :EventAttendance, source: 'user'
+
+  def attend?(user_id)
+    self.EventAttendance.where(user_id: user_id).exists?
```

Comment:
- [ ] **[OPTIONAL]** we can also check with:
```ruby
def attend?(user_id)
  attended_user_ids.include?(user_id)
end
```
This is just more readable. Note that `attended_user_ids` is a method generated from the `has_many :attended_users` association.